File: Behavioral/State/JavaScript/greenLight.js
// greenLight.js

const TrafficLightState = require("./trafficLightState");

class GreenLight extends TrafficLightState {
  handle(trafficLight) {
    console.log("Green Light - GO!");
    
    // FIX: Lazy import to avoid circular dependency
    const YellowLight = require("./yellowLight");
    trafficLight.setState(new YellowLight()); // Transition to Yellow
  }
}

module.exports = GreenLight;

File: Behavioral/State/Python/green_light.py
# green_light.py

from traffic_light_state import TrafficLightState
import yellow_light

class GreenLight(TrafficLightState):
  def handle(self, traffic_light):
    print("Green Light - GO!")
    traffic_light.set_state(yellow_light.YellowLight())  # Transition to Yellow
